R/match_pc_gpprac.R

```diff
@@ -0,0 +1,379 @@
+find_matching_data_by_chi <- function(data, match_variable) {
+  # Create all_match, the mean of the match variable, for those chis that have
+  # some matched and some unmatched
+  return_data <- data %>%
+    dplyr::group_by(chi) %>%
+    dplyr::summarise(all_match = mean({{ match_variable }})) %>%
+    dplyr::ungroup()
+
+  return(return_data)
+}
+
+#' Use situations where some records for a chi have a postcode to fill
+#' those that are missing
+#'
+#' @param data A data frame
+#' @param matching_data A data frame processed through [find_matching_data_by_chi()]
+#' @param type Either pc for pastcodes or gp for GP practices
+#'
+#' @return A data frame with some missing data filled based on type
+fill_data_from_chi <- function(data, matching_data, type = c("pc", "gp")) {
+  if (type == "pc") {
+    return_data <-
+      dplyr::left_join(data,
+        matching_data,
+        by = "chi"
+      ) %>%
+      # If the chi isn't missing and all_match isn't exactly 0 or 1,
+      # we can potentially fill the postcodes for that chi
+      dplyr::filter(!is_missing(chi) & !is.integer(all_match)) %>%
+      dplyr::group_by(chi) %>%
+      # Arrange by one of the keydates so the most recent postcode is at the top even if it's NA
+      # Then fill the values of postcode upwards so the NA is filled in
+      dplyr::arrange(
+        dplyr::desc(is.na(postcode)),
+        dplyr::desc(keydate2_dateformat),
+        .by_group = TRUE
+      ) %>%
+      # "NK010AA" is a known dummy postcode, so we replace this with NA
+      dplyr::mutate(postcode = dplyr::if_else(postcode == "NK010AA",
+        NA_character_,
+        postcode
+      )) %>%
+      # Now we can fill the postcodes for these chis
+      tidyr::fill(postcode, .direction = "up") %>%
+      dplyr::ungroup()
+  } else if (type == "gp") {
+    return_data <- dplyr::left_join(data_gpprac_match, data_for_matching, by = "chi") %>%
+      dplyr::filter(!is_missing(chi) & !is.integer(all_match)) %>%
+      dplyr::group_by(chi) %>%
+      # Arrange by one of the keydates so the most recent gpprac is at the top even if it's NA
+      # Then fill the values of gpprac upwards so the NA is filled in
+      dplyr::arrange(
+        dplyr::desc(is.na(gpprac)),
+        dplyr::desc(keydate2_dateformat),
+        .by_group = TRUE
+      ) %>%
+      dplyr::mutate(gpprac = dplyr::if_else(gpprac == 99999L, NA_real_, gpprac)) %>%
+      tidyr::fill(postcode, .direction = "up") %>%
+      dplyr::ungroup()
+  }
+
+  return(return_data)
+}
+
+#' Use the postcode from the most recent contact with an
+#' individual to fill missing data
+#'
+#' @param data A data frame
+#' @type Either pc for postcodes or gp for GP practices
+#'
+#' @return A data frame with some missing data filled
+<<<<<<< HEAD
+fill_data_from_keydate <- function(data, type = c("pc", "gp")) {
+=======
+fill_data_from_keydate <- function(data, var_to_fill, type = c("pc", "gp")) {
+>>>>>>> 3e001fde84178abc0abe8e038bb6952e69a1111c
```

Comment:
I'm not sure which conflict to follow, @shintoLamp can you advise what the latest line of code should be?